public/main.js

```diff
@@ -124,6 +131,7 @@ function reloadTodoList() {
         } else {
             clearCompleteButton.style.visibility = "hidden";
         }
+        todoListRefreshTimerId = setTimeout(reloadTodoList, todoListRefreshRate);
```

Comment:
If you are going with that route, I am not sure why you clear the timeout on line 94.
I would still suggest you use setInterval. 4 seconds is more than enough time for the server to respond and for your page to update itself. 
If you want to cover failure cases, I would catch any non 200 response from the server and clear the interval at that point.